dev/com.ibm.ws.ejbcontainer.timer.persistent_fat/fat/src/com/ibm/ws/ejbcontainer/timer/persistent/fat/tests/NoTablesTest.java

```diff
@@ -0,0 +1,91 @@
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.EnterpriseArchive;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.ShrinkHelper;
+
+import componenttest.annotation.ExpectedFFDC;
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.FATServletClient;
+
+/**
+ * Test ejbPersistentTimer feature when a datasource tables have not been created.
+ */
+@RunWith(FATRunner.class)
+public class NoTablesTest extends FATServletClient {
+
+    private static final String CNTR0218E = "CNTR0218E.* " + "NoDBPersistAutoTimerEJB.jar";
+    private static final String CNTR4002E = "CNTR4002E.* " + "NoDBPersistAutoTimerEJB.jar.* " + "NoDBPersistAutoTimerApp";
+    private static final String CWWKZ0106E = "CWWKZ0106E.* " + "NoDBPersistAutoTimerApp";
+    private static final String CWWKZ0002E = "CWWKZ0002E.* " + "NoDBPersistAutoTimerApp";
+
+    @Server("com.ibm.ws.ejbcontainer.timer.persistent.fat.NoTablesServer")
+    public static LibertyServer server;
+
+    @After
+    public void cleanUp() throws Exception {
+        // clean up just in case the test fails and doesn't complete server stop.
+        if (server != null && server.isStarted()) {
+            server.stopServer();
+        }
+    }
+
+    protected void runTest(String servlet, String testName) throws Exception {
+        FATServletClient.runTest(server, servlet, testName);
+    }
+
+    /**
+     * Test that an application with persistent automatic EJB timers will not start with
+     * appropriate errors in the logs when the ejbPersistentTimer feature is enabled, but
+     * no tables have been created in the datasource.
+     */
+    @Test
+    @ExpectedFFDC({ "javax.persistence.PersistenceException", "com.ibm.ws.exception.RuntimeWarning", "com.ibm.ws.container.service.state.StateChangeException" })
+    public void testNoTablesPersistentAutoTimer() throws Exception {
+        // Start the server with no applications
+        server.startServer();
+        server.setMarkToEndOfLog();
+
+        // Use ShrinkHelper to build the ear and export to server dropins
+        JavaArchive NoDBPersistAutoTimerEJB = ShrinkHelper.buildJavaArchive("NoDBPersistAutoTimerEJB.jar", "com.ibm.ws.ejbcontainer.timer.nodb.pauto.ejb.");
+        EnterpriseArchive NoDBPersistAutoTimerApp = ShrinkWrap.create(EnterpriseArchive.class, "NoDBPersistAutoTimerApp.ear");
+        NoDBPersistAutoTimerApp.addAsModule(NoDBPersistAutoTimerEJB);
+        ShrinkHelper.exportToServer(server, "dropins", NoDBPersistAutoTimerApp);
+
+        // Verify the application failed to start with correct messages
+        assertNotNull(CNTR0218E, server.waitForStringInLogUsingMark(CNTR0218E)); // persistent timers not supported
```

Comment:
```
* Wait for the specified regexp in the default logs from the last mark.
* The offset is also incremented every time this method is called.
```

All the exceptions are probably printed out in this order but if they ever print out in a different order the test will fail, I'd consider using the not mark version

Reply:
Actually, the 'mark' version is the version to use if the order may change.  waitForStringInLogUsingLastOffset will increment the offset with each call and force the order. waitForStringInLogUsingMark updates the last offset, but NOT the 'mark'.... so will search from last mark, which will be beginning of log for this test.  

For this particular scenario, the messages should always be in this order; and while switching to the Offset version of method would enforce the order, that approach seems to be deprecated in preference of using the Mark methods.